packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/types/model/FlutterSerializedModel.kt

```diff
@@ -23,12 +23,11 @@ data class FlutterSerializedModel(val serializedModel: SerializedModel) {
 
     fun toMap(): Map<String, Any> {
 
-        val cleanedSerializedData: Map<String, Any?> = serializedData.filterValues { it != null }
+        val cleanedSerializedData: Map<String, Any> = serializedData.filterValues { it != null } as Map<String, Any>
 
-        return mapOf(
-            "serializedData" to cleanedSerializedData,
-            "modelName" to modelName
-        )
+        val modelNameMap = mapOf("modelName" to modelName)
+
+        return cleanedSerializedData+modelNameMap
```

Comment:
Hum, so the `modelName` key-value pair will blend in model fields and values? 
If I have a model schema

```
type AModel @model {
  id: ID!
  modelName: Int!
}
```

The `Int` value would be overridden, correct?